src/main/java/org/threeten/extra/MutableClock.java

```diff
@@ -0,0 +1,511 @@
+        this.zone = zone;
+    }
+
+    /**
+     * Overrides the instant of this clock with the specified value.
+     *
+     * @param instant the new instant for this clock, not null
+     */
+    public void setInstant(Instant instant) {
+        Objects.requireNonNull(instant, "instant");
+        instantHolder.set(instant);
+    }
+
+    /**
+     * Adds the specified amount to this clock.
+     * <p>
+     * Atomically updates this clock to the value of the following expression:
+     * <pre>
+     *   ZonedDateTime.now(thisClock)
+     *                .plus(amountToAdd)
+     *                .toInstant()
+     * </pre>
+     *
+     * @param amountToAdd the amount to add, not null
+     * @throws DateTimeException if the addition cannot be made
+     * @throws ArithmeticException if numeric overflow occurs
+     * @see ZonedDateTime#plus(TemporalAmount)
+     */
+    public void add(TemporalAmount amountToAdd) {
+        Objects.requireNonNull(amountToAdd, "amountToAdd");
+        synchronized (instantHolder) {
+            ZonedDateTime current = ZonedDateTime.ofInstant(instantHolder.get(), zone);
+            ZonedDateTime result = current.plus(amountToAdd);
+            instantHolder.set(result.toInstant());
+        }
+    }
+
+    /**
+     * Adds the specified amount to this clock.
+     * <p>
+     * Atomically updates this clock to the value of the following expression:
+     * <pre>
+     *   ZonedDateTime.now(thisClock)
+     *                .plus(amountToAdd, unit)
+     *                .toInstant()
+     * </pre>
+     *
+     * @param amountToAdd the amount of the specified unit to add, not null
+     * @param unit the unit of the amount to add, not null
+     * @throws DateTimeException if the unit cannot be added
+     * @throws UnsupportedTemporalTypeException if the unit is not supported
+     * @throws ArithmeticException if numeric overflow occurs
+     * @see ZonedDateTime#plus(long, TemporalUnit)
+     */
+    public void add(long amountToAdd, TemporalUnit unit) {
+        Objects.requireNonNull(unit, "unit");
+        synchronized (instantHolder) {
+            ZonedDateTime current = ZonedDateTime.ofInstant(instantHolder.get(), zone);
+            ZonedDateTime result = current.plus(amountToAdd, unit);
+            instantHolder.set(result.toInstant());
+        }
+    }
+
+    /**
+     * Subtracts the specified amount from this clock.
+     * <p>
+     * Atomically updates this clock to the value of the following expression:
+     * <pre>
+     *   ZonedDateTime.now(thisClock)
+     *                .minus(amountToSubtract)
+     *                .toInstant()
+     * </pre>
+     *
+     * @param amountToSubtract the amount to subtract, not null
+     * @throws DateTimeException if the subtraction cannot be made
+     * @throws ArithmeticException if numeric overflow occurs
+     * @see ZonedDateTime#minus(TemporalAmount)
+     */
+    public void subtract(TemporalAmount amountToSubtract) {
```

Comment:
Given the warning that the clock should only go forwards, and the ability to do subtraction by passing in negative values to the add methods, I don't think the subtract methods make sense.